.danger/__tests__/changesets-that-need-core-check.spec.ts

```diff
@@ -0,0 +1,115 @@
+        './.danger/__fixtures__/changeset/popular-cheetahs-punch.md',
+        './.danger/__fixtures__/changeset/pretty-cameras-burn.md',
+      ])
+    ).toEqual([
+      './.danger/__fixtures__/changeset/popular-cheetahs-punch.md',
+      './.danger/__fixtures__/changeset/pretty-cameras-burn.md',
+    ]);
+  });
+});
+
+describe('changesetsThatNeedCoreCheck()', () => {
+  beforeEach(() => {
+    global.warn = jest.fn();
+    global.message = jest.fn();
+    global.fail = jest.fn();
+    global.markdown = jest.fn();
+  });
+
+  afterEach(() => {
+    global.warn = undefined;
+    global.message = undefined;
+    global.fail = undefined;
+    global.markdown = undefined;
+    global.danger = undefined;
+  });
+
+  it('should fail for each modified changeset that does not tag core', () => {
+    global.danger = {
+      git: {
+        modified_files: [
+          './.danger/__fixtures__/changeset/heavy-peaches-repeat.md',
+          './.danger/__fixtures__/changeset/pink-masks-walk.md',
+          './.danger/__fixtures__/changeset/popular-cheetahs-punch.md',
+          './.danger/__fixtures__/changeset/pretty-cameras-burn.md',
+        ],
+        created_files: [],
+      },
+    };
+    changesetsThatNeedCoreCheck();
+    expect(global.fail).toHaveBeenCalledTimes(2);
+  });
+
+  it('should fail for each created changeset that does not tag core', () => {
+    global.danger = {
+      git: {
+        modified_files: [],
+        created_files: [
+          './.danger/__fixtures__/changeset/pink-masks-walk.md',
+          './.danger/__fixtures__/changeset/pretty-cameras-burn.md',
+        ],
+      },
+    };
+    changesetsThatNeedCoreCheck();
+    expect(global.fail).toHaveBeenCalledTimes(1);
+  });
+
+  it('should fail for each modified and created changeset that does not tag core', () => {
+    global.danger = {
+      git: {
+        modified_files: [
+          './.danger/__fixtures__/changeset/heavy-peaches-repeat.md',
+          './.danger/__fixtures__/changeset/pink-masks-walk.md',
+          './.danger/__fixtures__/changeset/pretty-cameras-burn.md',
+        ],
+        created_files: ['./.danger/__fixtures__/changeset/popular-cheetahs-punch.md'],
+      },
+    };
+    changesetsThatNeedCoreCheck();
+    expect(global.fail).toHaveBeenCalledTimes(2);
+  });
+
+  it('should name the failing changeset that does not tag core', () => {
+    global.danger = {
+      git: {
+        modified_files: [
+          './.danger/__fixtures__/changeset/heavy-peaches-repeat.md',
+          './.danger/__fixtures__/changeset/pink-masks-walk.md',
+        ],
+        created_files: ['./.danger/__fixtures__/changeset/popular-cheetahs-punch.md'],
+      },
+    };
+    changesetsThatNeedCoreCheck();
+    expect(global.fail).toHaveBeenCalledWith(
+      'Changeset ./.danger/__fixtures__/changeset/popular-cheetahs-punch.md does not also apply to Paste Core. Please include core with this changeset.'
```

Comment:
Saying "Changeset ___ does not also apply to Paste Core." confused me, since it made me feel like this check was saying "You've applied this to Core and it shouldn't be there". How about "Changeset ___ does not mention a change to Paste Core."?

Reply:
Into it